component/dependencies/cloudscale-metrics-collector/component/main.jsonnet

```diff
@@ -0,0 +1,108 @@
+local kap = import 'lib/kapitan.libjsonnet';
+local inv = kap.inventory();
+local params = inv.parameters.cloudscale_metrics_collector;
+
+local namespace = 'appuio-cloud-reporting';
+local image = 'ghcr.io/vshn/cloudscale-metrics-collector:v0.2.0';
+
+
+local labels = {
+  'app.kubernetes.io/name': 'appuio-cloud-reporting',
+  'app.kubernetes.io/managed-by': 'commodore',
+  'app.kubernetes.io/part-of': 'syn',
+};
+
+
+{
+  namespace: {
+    kind: 'Namespace',
+    apiVersion: 'v1',
+    metadata: {
+      name: namespace,
+      labels+: labels,
+    },
+  },
+
+  secret: {
+    assert params.cloudscale.token != null : 'cloudscale.token must be set.',
+    kind: 'Secret',
+    apiVersion: 'v1',
+    metadata: {
+      name: 'cloudscale-api',
+      namespace: namespace,
+      labels+: labels,
+    },
+    stringData: {
+      token: params.cloudscale.token
+    }
+  },
```

Comment:
I suggest to have a look at this page about handling secrets: https://syn.tools/syn/explanations/commodore-components/secrets.html

That being said, the token should be managed via Vault, and not be stored as a plain text variable in the hierarchy.

Reply:
> page about handling secrets
I've changed the implementation to use this.

> the token should be managed via Vault, 
As far as I understand it this is the responsibility of the user of the component, not the component itself, correct?